javascript/ql/src/semmle/javascript/TypeScript.qll

```diff
@@ -1315,6 +1315,15 @@ class TypeAssertion extends Expr, @typeassertion {
   override ControlFlowNode getFirstControlFlowNode() {
     result = getExpression().getFirstControlFlowNode()
   }
+
+  override Expr getUnderlyingValue() {
+    result = getExpression().getUnderlyingValue()
+  }
+
+  override Expr getUnderlyingReference() {
```

Comment:
I'm not sure type assertions can be used as LHSs; @asger-semmle?

Reply:
Didn't think so either, but it seems to work.